src/main/java/io/supertokens/webserver/api/oauth/OAuthAuthAPI.java

```diff
@@ -46,26 +48,26 @@ public OAuthAuthAPI(Main main) {
         super(main, RECIPE_ID.OAUTH.toString());
     }
 
+    private static final List<String> REQUIRED_FIELDS_FOR_POST = Arrays.asList(new String[]{"clientId", "scope",
+            "state", "redirectUri", "responseType"});
+
     @Override
     public String getPath() {
         return "/recipe/oauth/auth";
     }
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+
         JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
-        String clientId = InputParser.parseStringOrThrowError(input, "clientId", false);
-        String redirectUri = InputParser.parseStringOrThrowError(input, "redirectUri", false);
-        String responseType = InputParser.parseStringOrThrowError(input, "responseType", false);
-        String scope = InputParser.parseStringOrThrowError(input, "scope", false);
-        String state = InputParser.parseStringOrThrowError(input, "state", false);
+        InputParser.collectAllMissingRequiredFieldsAndThrowError(input, REQUIRED_FIELDS_FOR_POST);
```

Comment:
should we normalise the input as well? for example, making clientId all lower case and trimming it

Reply:
Well, based on how the other InputParser methods are working, I would say no. The other inputParser methods are not changing the given input.
Regarding the aspect of sending a possible wrong value to hydra: I think hydra should handle it (and for example: sending a non-existent clientId will result in an error from hydra).
I don't think we should do any - possibly unexpected - changes to the input. In my opinion, we only should "translate" the property keys, to be able to communicate with hydra, and that's it. If the sdk gives a wrong input that's an error on the sdk part and the core handles it accordingly.